### ISSUE TITLE
Crash when using Microsoft.NET.Test.Sdk 15.6.0 on net461

### ISSUE THREAD

**Issue by altso** [nunit/nunit3-vs-adapter#468]
I get an error with the most recent version of `Microsoft.NET.Test.Sdk` `15.6.0` on `net461`. 

The error disappears when I downgrade to `Microsoft.NET.Test.Sdk` `15.5.0` or change the framework to `netcoreapp2.0`.

I also tried `xUnit` with `Microsoft.NET.Test.Sdk` `15.6.0` and it works just fine.

.csproj file:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
    <IsPackable>false</IsPackable>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="nunit" Version="3.9.0" />
    <PackageReference Include="NUnit3TestAdapter" Version="3.9.0" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.6.0" />
  </ItemGroup>
</Project>
```

Sample output:
```
>dotnet --version
2.0.2

>dotnet test
Build started, please wait...
Build completed.

Test run for D:\Temp\NUnitIssue\bin\Debug\net461\NUnitIssue.dll(.NETFramework,Version=v4.6.1)
Microsoft (R) Test Execution Command Line Tool Version 15.3.0-preview-20170628-02
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
NUnit Adapter 3.9.0.0: Test execution started
Running all tests in D:\Temp\NUnitIssue\bin\Debug\net461\NUnitIssue.dll
Exception System.InvalidOperationException, Exception thrown executing tests in D:\Temp\NUnitIssue\bin\Debug\net461\NUnitIssue.dll
Operation is not valid due to the current state of the object.
   at Mono.Cecil.ModuleDefinition.ProcessDebugHeader()
   at Mono.Cecil.ModuleDefinition.ReadSymbols(ISymbolReader reader)
   at Mono.Cecil.ModuleReader.ReadSymbols(ModuleDefinition module, ReaderParameters parameters)
   at Mono.Cecil.ModuleReader.CreateModuleFrom(Image image, ReaderParameters parameters)
   at Mono.Cecil.ModuleDefinition.ReadModule(Stream stream, ReaderParameters parameters)
   at Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters)
   at NUnit.VisualStudio.TestAdapter.NavigationDataProvider.CacheNewTypes(String assemblyPath, IDictionary`2 types)
   at NUnit.VisualStudio.TestAdapter.NavigationDataProvider.GetNavigationData(String className, String methodName)
   at NUnit.VisualStudio.TestAdapter.TestConverter.MakeTestCaseFromXmlNode(XmlNode testNode)
   at NUnit.VisualStudio.TestAdapter.TestConverter.ConvertTestCase(XmlNode testNode)
   at NUnit.VisualStudio.TestAdapter.NUnit3TestExecutor.RunAssembly(String assemblyPath, TestFilter filter)
NUnit Adapter 3.9.0.0: Test execution complete
No test is available in D:\Temp\NUnitIssue\bin\Debug\net461\NUnitIssue.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.

```

**Comment by rprouse** [comment]
You need to add `<DebugType>Full</DebugType>` to the first `PropertyGroup` for .NET 4.x tests. The full framework version of the NUnit adapter doesn't support the new PDB type yet.

Let me know if this fixes it for you. Marking as confirm until then.

**Comment by rprouse** [comment]
Odd that it works with 15.5.0 though, so I might be wrong 😄

**Comment by jnm2** [comment]
This is the same stack trace fixed in https://github.com/nunit/nunit3-vs-adapter/pull/454. If you want to help us be sure we've covered your scenario (which we appreciate!), you can merge this with your nuget.config and try version `3.10.0-dev-00750` or newer:

```xml
﻿<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <packageSources>
        <add key="NUnit 3 VS adapter prerelease" value="https://www.myget.org/F/nunit/api/v3/index.json" />
    </packageSources>
</configuration>
```

**Comment by altso** [comment]
Thank you. I will check `DebugType` and Dev feed tomorrow

**Comment by jnm2** [comment]
@altso Thank you! The DebugType workaround should work immediately for you. While trying out the dev feed, it would be good to test without that workaround because our hope is that any DebugType should work in 3.10:

- full (Windows debug info is put in DLL)
- pdbonly (Windows debug info is put in PDB)
- embedded (portable debug info is put in DLL)
- portable (portable debug info is put in PDB)

**Comment by altso** [comment]
@jnm2, @rprouse, after some additional testing it appears that **any** of the following resolves the issue:

1. Adding `<DebugType>Full</DebugType>` to `.csproj` file.
2. Updating `NUnit3TestAdapter` to the latest dev version (I tried `3.10.0-dev-00820`).
3. Updating `dotnet sdk` to `2.1.4`. I did not try anything between `2.0.2` and `2.1.4`. The original issue was submitted for `2.0.2`.

Since there are multiple options to fix the error, I'm closing the issue. Thank you!
